src/Xamarin.Android.Build.Tasks/Utilities/ManifestDocument.cs

```diff
@@ -336,16 +336,20 @@ public IList<string> Merge (TaskLoggingHelper log, TypeDefinitionCache cache, Li
 				if (PackageName == null)
 					PackageName = t.Namespace;
 
-				var name        = JavaNativeTypeManager.ToJniName (t, cache).Replace ('/', '.');
-				var compatName  = JavaNativeTypeManager.ToCompatJniName (t, cache).Replace ('/', '.');
-				if (((string) app.Attribute (attName)) == compatName) {
-					app.SetAttributeValue (attName, name);
+				string name, compatName;
```

Comment:
Instead of declaring w/o initializing, I think it would be better to "double declare" them, once within the following `if` block, and separately *within* the `try` block:

```csharp
if (t.IsSubclassOf ("Android.App.Application", cache)) {
    var (name, compatName) = GetNames (t, cache);
    // …
}
…
try {
    var (name, compatName) = GetNames (t, cache);
    // …
}
```

Reply:
I put `(string name, string compatName) = GetNames (t, cache)` so they types are more clear.